cyral/resource_cyral_repository_user_account.go

```diff
@@ -0,0 +1,414 @@
+				ForceNew:    true,
+			},
+
+			"name": {
+				Description: "The name of the database account.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+
+			"auth_scheme": {
+				Description: "Credential option. List of supported types: " +
+					supportedTypesMarkdown(allAuthSchemes),
+				Required: true,
+				Type:     schema.TypeSet,
+				MaxItems: 1,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"environment_variable": {
+							Description: "Credential option to set the repository user account from " +
+								"Environment Variable.",
+							Optional: true,
+							Type:     schema.TypeSet,
+							MaxItems: 1,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"variable_name": {
+										Description: "Name of the environment variable that will store credentials.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+								},
+							},
+						},
+						"aws_iam": {
+							Description: "Credential option to set the repository user account from " +
+								"AWS IAM.",
+							Optional: true,
+							Type:     schema.TypeSet,
+							MaxItems: 1,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"role_arn": {
+										Description: "The AWS IAM roleARN to gain access to the database.",
+										Type:        schema.TypeString,
+										Required:    true,
+										Sensitive:   true,
+									},
+								},
+							},
+						},
+						"aws_secrets_manager": {
+							Description: "Credential option to set the repository user account from " +
+								"AWS Secrets Manager.",
+							Optional: true,
+							Type:     schema.TypeSet,
+							MaxItems: 1,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"secret_arn": {
+										Description: "The AWS Secrets Manager secretARN to gain access to the database.",
+										Type:        schema.TypeString,
+										Required:    true,
+										Sensitive:   true,
+									},
+								},
+							},
+						},
+						"cyral_storage": {
+							Description: "Credential option to set the repository user account from " +
+								"Cyral Storage.",
+							Optional: true,
+							Type:     schema.TypeSet,
+							MaxItems: 1,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"password": {
+										Description: "The Cyral Storage password to gain access to the database.",
+										Type:        schema.TypeString,
+										Required:    true,
+										Sensitive:   true,
+									},
+								},
+							},
+						},
+						"hashicorp_vault": {
+							Description: "Credential option to set the repository user account from " +
+								"Hashicorp Vault.",
+							Optional: true,
+							Type:     schema.TypeSet,
+							MaxItems: 1,
+							MinItems: 1,
```

Comment:
Is this actually correct here even though for Vault we have 2 fields ("path" and "is_dynamic_user_account") ?